airflow/api_fastapi/serializers/dags.py

```diff
@@ -0,0 +1,89 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from datetime import datetime
+from typing import Any
+
+from itsdangerous import URLSafeSerializer
+from pydantic import (
+    BaseModel,
+    computed_field,
+    field_validator,
+)
+
+from airflow.configuration import conf
+from airflow.serialization.pydantic.dag import DagTagPydantic
+
+
+class DAGModelResponse(BaseModel):
```

Comment:
This is very close to `DagModelPydantic` that we have for AIP-44. But cannot really re-use it because fields required for the public API are a bit different.

Also if I am not mistaken that will get removed once we do the Task Interface AIP, so I didn't bother too much trying to factorize that code.

Reply:
Which fields were missing? I'm happy to move some of these details to the `{dag}/details` endpoint. and keep our list response simpler.